consistenthash/consistenthash.go

```diff
@@ -19,24 +19,32 @@ package consistenthash
 
 import (
 	"hash/crc32"
+	"math/bits"
 	"sort"
 	"strconv"
 )
 
 type Hash func(data []byte) uint32
 
 type Map struct {
-	hash     Hash
-	replicas int
-	keys     []int // Sorted
-	hashMap  map[int]string
+	hash                 Hash
+	replicas             int
+	prefixTableExpansion int
+
+	keys    []int // Sorted
+	hashMap map[int]string
+
+	prefixBits  uint32
+	prefixShift uint32
+	prefixTable []string
 }
 
-func New(replicas int, fn Hash) *Map {
+func New(replicas int, tableExpansion int, fn Hash) *Map {
```

Comment:
This is a breaking API change for anybody using this package. Also, this parameter isn't documented.

Maybe make a new NewXxx func instead taking the new parameter and make the existing `New` default to 6 or whatever?